src/util/samplebuffer.h

```diff
@@ -34,64 +34,65 @@ namespace mixxx {
 //
 class SampleBuffer final {
   public:
-    SampleBuffer()
+    constexpr SampleBuffer()
         : m_data(nullptr),
           m_size(0) {
     }
     explicit SampleBuffer(SINT size);
     SampleBuffer(SampleBuffer&) = delete;
     SampleBuffer(SampleBuffer&& that)
-        : m_data(that.m_data),
-          m_size(that.m_size) {
-        that.m_data = nullptr;
-        that.m_size = 0;
+            : m_data(std::exchange(that.m_data, nullptr)),
+              m_size(std::exchange(that.m_size, 0)) {
     }
-    virtual ~SampleBuffer() final;
+    ~SampleBuffer();
 
     SampleBuffer& operator=(SampleBuffer& that) = delete;
-    SampleBuffer& operator=(SampleBuffer&& that) {
-        swap(that);
-        return *this;
-    }
+    SampleBuffer& operator=(SampleBuffer&& that);
 
-    constexpr SINT size() const {
+    constexpr SINT size() const noexcept {
         return m_size;
     }
 
-    CSAMPLE* data(SINT offset = 0) {
+    constexpr CSAMPLE* data() noexcept {
+        return m_data;
+    }
+    constexpr const CSAMPLE* data() const noexcept {
+        return m_data;
+    }
```

Comment:
in the future we should mark these `[[deprecated]]` because the span API is just better, but that is out-of-scope for this PR.

Reply:
Actually the span isn't a good option too.

I suggest a custom implemented span, which also keeps the following info:
- samplerate of the buffer
- channel count
- frame count (instead of simply size)

This isn't essential, but very handy in many situations, if you want to add assertions or you're debugging the app and want to understand the program while in the debugger.

Reply:
Yes, that will come in the future. We want an abstraction based on `mdspan` that keeps track of all that, but the current API does not offer anything better than `span` so we might as well use that.

Reply:
Just use `SampleBuffer::ReadSlice`, etc... so that way when you add more info to the sample buffer they'll be naturally present in the read slice.

Looking at `mdspan` I don't see it better than ReadSlice or WriteSlice.